crates/bevy_math/src/ray_cast/dim2/capsule.rs

```diff
@@ -0,0 +1,193 @@
+use crate::prelude::*;
+
+// This is mostly the same as `Capsule3d`, but with 2D types.
+impl PrimitiveRayCast2d for Capsule2d {
+    #[inline]
+    fn local_ray_cast(&self, ray: Ray2d, max_distance: f32, solid: bool) -> Option<RayHit2d> {
+        // Adapted from Inigo Quilez's ray-capsule intersection algorithm: https://iquilezles.org/articles/intersectors/
+
+        let radius_squared = self.radius * self.radius;
+
+        let ba = 2.0 * self.half_length;
+        let oa = Vec2::new(ray.origin.x, ray.origin.y + self.half_length);
+
+        let baba = ba * ba;
+        let bard = ba * ray.direction.y;
+        let baoa = ba * oa.y;
+        let rdoa = ray.direction.dot(oa);
+        let oaoa = oa.dot(oa);
+
+        // Note: We use `f32::EPSILON` to avoid division by zero later for rays parallel to the capsule's axis.
+        let a = (baba - bard * bard).max(f32::EPSILON);
+        let b = baba * rdoa - baoa * bard;
+        let c = baba * oaoa - baoa * baoa - radius_squared * baba;
```

Comment:
You can skip a multiplication by changing this to:
```suggestion
        let c = baba * (oaoa - radius_squared) - baoa * baoa;
```
Not sure how this affects numerical stability though, I assume not much.
Also I wonder if perhaps the paren is not actually good for perf.
Just an idea.